src/reactA11yImgHasAltRule.ts

```diff
@@ -99,6 +104,8 @@ class ImgHasAltWalker extends Lint.RuleWalker {
                 getFailureStringNoAlt(tagName)
             );
         } else {
+            const altAttributeValue = getStringLiteral(altAttribute);
+            const hasImageInAlt: boolean = !!String(altAttributeValue).toLowerCase().match(/(\b|[_-])(photo|image|picture)(\b|[_-])/);
```

Comment:
This is some fun JS syntax 🙂 but IMO the error message would look a little cleaner if it only said the redundant descriptor(s) actually in the tag. Instead of a single regexp, could you please have it collect the list and print only the relevant ones?

Reply:
> really new to programming

...and you're working on an open source project that uses the TypeScript APIs to do static analysis? Great start! 👏

> is the single regex to complex in any form?

Sorry, I don't follow what you mean by that. Are you referring to just here, or regular expressions in general, or...?